pandas/core/arrays/arrow/array.py

```diff
@@ -695,22 +695,23 @@ def _evaluate_op_method(self, other, op, arrow_funcs):
         other = self._box_pa(other)
 
         if pa.types.is_string(pa_type) or pa.types.is_binary(pa_type):
-            if op in [operator.add, roperator.radd, operator.mul, roperator.rmul]:
+            if op in [operator.add, roperator.radd]:
                 sep = pa.scalar("", type=pa_type)
                 if op is operator.add:
                     result = pc.binary_join_element_wise(self._pa_array, other, sep)
                 elif op is roperator.radd:
                     result = pc.binary_join_element_wise(other, self._pa_array, sep)
-                else:
-                    if not (
-                        isinstance(other, pa.Scalar) and pa.types.is_integer(other.type)
-                    ):
-                        raise TypeError("Can only string multiply by an integer.")
-                    result = pc.binary_join_element_wise(
-                        *([self._pa_array] * other.as_py()), sep
-                    )
                 return type(self)(result)
-
+            elif op in [operator.mul, roperator.rmul]:
+                result = type(self)._evaluate_binary_repeat(self._pa_array, other)
+                return type(self)(result)
+        elif (
```

Comment:
Can you put this in the same elif statement?

```
elif multiplication
    if one argument is string or binary:
        do repeat
```

Reply:
Sure, although I think it would need to be an if (as opposed to elif) since the first check would be hit in case of string multiplied by int, but falls through all inner if conditions. Since this should support string * int and int * string, I think we'd need a separate check to handle both scenarios.

Reply:
The issue with that approach, is we have the first condition to check if source series is binary type, which handles addition case, I believe we need 2 separate conditions, because the repeat should be invoked for string * int and int * string, the first if condition would handle the first case, the 2nd if condition for the 2nd case.

Reply:
OK yeah that makes sense then